async_fuse/src/main.rs

```diff
@@ -44,10 +44,146 @@ use log::debug;
 mod fuse;
 mod memfs;
 pub mod util;
+use common::error::Context;
+use common::etcd_delegate::EtcdDelegate;
 use fuse::session::Session;
+use std::env;
 
+/// Ip address environment var name
+const ENV_IP_ADDRESS: &str = "pod_ip_address";
 /// Argument name of FUSE mount point
 const MOUNT_POINT_ARG_NAME: &str = "mountpoint";
+/// Argument name of ETCD addresses
+const ETCD_ADDRESS_ARG_NAME: &str = "etcd";
+/// Argument name of Volume information
+const VOLUME_INFO: &str = "volume_info";
+/// ETCD node id lock
+const ETCD_NODE_ID_LOCK: &str = "datenlord_etcd_node_id_lock";
+/// ETCD node id counter key
+const ETCD_NODE_ID_COUNTER_KEY: &str = "datenlord_etcd_node_id_counter";
+/// ETCD ndoe id info
+const ETCD_NODE_ID_INFO_PREFIX: &str = "datenlord_etcd_node_id_info_";
+/// ETCD volume information lock
+const ETCD_VOLUME_INFO_LOCK: &str = "datenlord_etcd_volume_info_lock";
+/// ETCD volume information prefix
+const ETCD_VOLUME_INFO_PREFIX: &str = "datenlord_etcd_volume_info_";
+
+/// Register current node to etcd and get a dedicated node id.
+/// The registered information contains IP.
+async fn register_node_id(etcd_client: &EtcdDelegate) -> anyhow::Result<u64> {
```

Comment:
Is this overlapped with https://github.com/datenlord/datenlord/blob/master/csi/src/meta_data.rs#L139?

Reply:
When will that function be called? And this function also register an IP.

Reply:
That is called when the pod is created.